Dockerfile

```diff
@@ -47,3 +58,14 @@ COPY --from=server_builder /weaviate-server /bin/weaviate
 RUN apk add --no-cache --upgrade ca-certificates openssl
 RUN mkdir ./modules
 CMD [ "--host", "0.0.0.0", "--port", "8080", "--scheme", "http"]
+
+###############################################################################
+# Weaviate with debug enabled and ready for remote debugging. Use for debugging/dev purpose only!
+FROM alpine AS weaviate_debug
+ENTRYPOINT [ "/bin/dlv" ]
+COPY --from=server_builder_debug /go/bin/dlv /bin/dlv
+COPY --from=grpc_health_probe_builder /bin/grpc_health_probe /bin/
+COPY --from=server_builder_debug /weaviate-server /bin/weaviate
+RUN apk add --no-cache --upgrade ca-certificates openssl
+RUN mkdir ./modules
+CMD ["--listen=:2345", "--headless=true", "--check-go-version=false", "--log=true", "--accept-multiclient", "--api-version=2", "exec", "/bin/weaviate", "--", "--host", "0.0.0.0", "--port", "8080", "--scheme", "http"]
```

Comment:
I would move this part to a separate `debug.Dockerfile` I would not mix this with `production` `Dockerfile`